rest_docs/rest-catalog-open-api-v0.1.yaml

```diff
@@ -0,0 +1,761 @@
+              value: "prod.accounting"
+        # TODO - There's a much more native way to handle pagination
+        - name: limit
+          description: number of values to return in one request
+          in: query
+          required: false
+          schema:
+            type: integer
+          example: 10
+        - name: offset
+          description: Place in the response to continue from if paginating
+          in: query
+          schema:
+            type: integer
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ListTablesResponse'
+    post:
+      tags:
+        - Catalog API
+      summary: Create a table with the identifier given in the body
+      operationId: createTable
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/CreateTableRequest'
+        required: true
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/CreateTableResponse'
+  /v1/tables/renameTable:
+    post:
+      tags:
+        - Catalog API
+      summary: Rename a table from its current name to a new name within the same catalog
+      description: Rename a table within the same catalog
+      operationId: renameTable
+      requestBody:
+        description: Current table identifier to rename and new table identifier to rename to
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/RenameTableRequest'
+        required: true
+      responses:
+        '200':
+          description: OK
+        '401':
+          description: Unauthorized
+        # TODO - Probably 404 will cause monitoring headaches as it's very hard to monitor if a client made
+        #        a valid request, but the table didn't exist, or if a client has been misconfigured or has
+        #        some other sort of bug and is calling endpoints that don't exist.
+        #        Need to settle on a non-404 error code.
+        '412':
+          description: Table to rename from does not exist
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/NoSuchTableError'
+              example:
+                '{ error: { message: "Table does not exist", type: "NoSuchTableException", code: 41202 }'
+        '409':
+          description: The new table identifier, the to table rename to, already exists.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/TableAlreadyExistsError'
+              example:
+                '{ error: { message: "Namespace already exists", type: "AlreadyExistsException", code: 40902 }'
+  /v1/namespaces/{namespace}/properties:
```

Comment:
why do we need a specific path for properties? I thought it could just be a part of the namespace response.

Reply:
I added this one to be able to mirror this API of `loadNamespaceMetadata`: https://github.com/apache/iceberg/blob/f96f06ea07841c137a98a25e4edf62c508475801/api/src/main/java/org/apache/iceberg/catalog/SupportsNamespaces.java#L89-L96

Right now, between the two interfaces `Catalog` and `SupportsNamespaces`, I don't think there's a method that just returns a namespace. Though a namespace at its minimum is really just a combination of its properties and its name.

In terms of the current API for returning the details of a namespace, the following exists:
1) listing the namespaces underneath a namespace (or all namespaces)
2) listing the tables underneath a namespace
3) getting the properties of a namespace.

The other reason I added the `properties` route is to be able to support `setProperties` and `removeProperties`: https://github.com/apache/iceberg/blob/f96f06ea07841c137a98a25e4edf62c508475801/api/src/main/java/org/apache/iceberg/catalog/SupportsNamespaces.java#L107-L129

I'm still very open to making changes that seem appropriate, especially as you have more experience writing catalogs than I do. What do you think knowing that we'd need to support updating / removing a subset of properties (e.g. `setProperties` and `removeProperties`)?

Reply:
Oh actually, I see. I have duplicated this functionality here in `GET /v1/namespaces/{namespace}` which is the same as this one. https://github.com/apache/iceberg/pull/3561/files#diff-111c5d0e1ee3097e79ed73a1546ec698a1ced352827c109c41ec21cf68db120eR441-R452

So the only thing we'd need to be able to support would be `setProperties` and `removeProperties`.

I will remove one or the other once we sort that out.

Reply:
I chose to remove the operation that was `getNamespace`, since its essentially the same response and there is not an interface for just getting a namespace (just `loadNamespaceMetadata`).

We do need some sort of `properties` route for `setProperties` and `removeProperties`, hence why I left it here.

Let me know if you have any concerns.

Reply:
Yeah, we should make this `GET /namespaces/{namespace}`. We can probably still use the `loadNamespaceMetadata` name. I like using the names that correspond to the interfaces.

Reply:
That works for me. I'm happy to move this one back to `GET /namespaces/{namespace}/properties` and then leave the `POST` and `PUT`.

Reply:
I'm going to resolve this discussion now that we've moved loadNamespaceMetadata outside of the `properties` route and have standardized the `setProperties` and `removeProperties` endpoints a bit.

Please feel free to reopen if you have concerns!

Reply:
Unresolving this comment, as elsewhere it was suggested to use DELETE for removing properties. However, DELETE can't have a request body.

I can't find that original comment thread so reviving it here.

What verb / route do you think would work best for `removeProperties` from `SupportsNamespace`, given that `DELETE` cannot have a request body? cc @rdblue @jackye1995

Reply:
I moved `removeProperties` back to `PUT` for now, which can have a request body.

Reply:
These have been moved into one POST call which essentially is `updateProperties`.